config/catalog.go

```diff
@@ -27,3 +38,40 @@ func (config *CatalogConfig) normalize(cofnigPath string) {
 		}
 	}
 }
+
+func ReadCatalogConfig(opts *options.TerragruntOptions) (*TerragruntConfig, error) {
```

Comment:
NIT: do you mind adding a comment here explaining what this method does and _why_?

Namely:

- We want users to be able to browse to any folder in an `infra-live` repo, run `terragrunt catalog` (with no URL) arg, and we automatically find the "closest" terragrunt.hcl, and load the catalogs from its `catalog` config.
- So this method searches the current working directory and its parent folders for a `terragrunt.hcl`.
- We then need to parse this file. Our normal parsing won't always work, as some `terragrunt.hcl` files are meant to be used from an `include` and/or they might use `find_in_parent_folders`, so they only work from certain child folders.
- So if we find a `terragrunt.hcl` that already has an `include`, we parse that as-is; if we find a `terragrunt.hcl` without `include`, we generate a stub child `terragrunt.hcl` in memory with an `include` to pull in the one we found.

Reply:
Of course, thanks for the example text. I will try not to miss important explanations of functions in the future.